### PR TITLE
Update farm card withdrawal and deposit modals

### PR DESCRIPTION
**Pull request by RichardPK**
- Creation of new `ModalInput` component to match the new modal designs
- Restyling `DepositModal` and `WithdrawModal` to match new designs and make use of new `ModalInput` component
- Move the `addLiquidityUrl` string construction logic out of the `ApyCalculatorModal` and the card `DetailsSection` - instead construct this within the `FarmCard` and pass it down.
- Added some new exchange URL consts to `config/index.ts`

### CODE REVIEW

File: src/components/ModalInput/ModalInput.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  align-items: center;
+  color: ${(props) => props.theme.colors.primary};
+  display: flex;
+  font-size: 14px;
+  font-weight: 700;
+  height: 44px;
+  justify-content: flex-end;
+`
+
+const StyledTokenSymbol = styled.span`
```
Comment (RabbitDoge) [inline src/components/ModalInput/ModalInput.tsx:61]: `<Text bold color="primary">...</Text>` makes the job

File: src/views/Farms/components/FarmCard/FarmCard.tsx [hunk 2]
```diff
@@ -122,6 +123,8 @@ const FarmCard: React.FC<FarmCardProps> = ({ farm, removed, cakePrice, bnbPrice,
   const farmAPY = farm.apy && farm.apy.times(new BigNumber(100)).toNumber().toLocaleString('en-US').slice(0, -1)
 
   const { quoteTokenAdresses, quoteTokenSymbol, tokenAddresses } = farm
+  const liquidityUrlPathParts = getLiquidityUrlPathParts({ quoteTokenAdresses, quoteTokenSymbol, tokenAddresses })
+  const addLiquidityUrl = `https://exchange.pancakeswap.finance/#/add/${liquidityUrlPathParts}`
```
Comment (RabbitDoge) [inline src/views/Farms/components/FarmCard/FarmCard.tsx:127]: It's probably a good idea to store the exchange base url in the config

Comment (RichardPK) [reply src/views/Farms/components/FarmCard/FarmCard.tsx:127]: Nice - added

File: src/components/ModalInput/ModalInput.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Text, Button, Input, InputProps, Flex, Link } from '@pancakeswap-libs/uikit'
+import useI18n from '../../hooks/useI18n'
+
+interface ModalInputProps {
+  max: string
```
Comment (RichardPK) [inline src/components/ModalInput/ModalInput.tsx:7]: This was previously `number | string` - do we know why? Having the potential for multiple types caused some problems with me trying to convert the string to a float and then round, later on in the component.

File: src/components/ModalInput/ModalInput.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  max,
+  symbol,
+  onChange,
+  onSelectMax,
+  value,
+  addLiquidityUrl,
+  withdrawal,
+}) => {
+  const TranslateString = useI18n()
+  const isBalanceZero = max === '0' || !max
```
Comment (RichardPK) [inline src/components/ModalInput/ModalInput.tsx:69]: This condition determines a few things, including visually whether to show the red 'failure' input outline, and let the user know they have no tokens to stake.

File: src/components/ModalInput/ModalInput.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+          <Button size="sm" onClick={onSelectMax} mr="8px">
+            {TranslateString(452, 'Max')}
+          </Button>
+          <Text fontSize="16px">{symbol}</Text>
+        </Flex>
+      </StyledTokenInput>
+      {isBalanceZero && (
+        <StyledErrorMessage fontSize="14px" color="failure">
+          No tokens to stake:{' '}
+          <Link fontSize="14px" bold={false} href={addLiquidityUrl} external color="failure">
```
Comment (RichardPK) [inline src/components/ModalInput/ModalInput.tsx:93]: Not sure if this was intended as a link but I thought hey why not let's drive dem clicks

File: src/components/ModalInput/ModalInput.tsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  ${({ theme }) => theme.mediaQueries.sm} {
+    width: auto;
+  }
+`
+
+const StyledErrorMessage = styled(Text)`
+  position: absolute;
+  bottom: -22px;
```
Comment (RichardPK) [inline src/components/ModalInput/ModalInput.tsx:53]: Safari sometimes throws its toys out with absolute styles... can someone validate the styles look OK? I have none of my crypto/metamask wallets setup on it.

File: src/components/ModalInput/ModalInput.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+  symbol: string
+  onSelectMax?: () => void
+  onChange: (e: React.FormEvent<HTMLInputElement>) => void
+  placeholder?: string
+  value: string
+  addLiquidityUrl?: string
+  withdrawal?: boolean
+}
+
+const getBoxShadow = ({ isWarning = false, theme }) => {
```
Comment (RichardPK) [inline src/components/ModalInput/ModalInput.tsx:17]: A copy & paste job from the uikit input - it could maybe be exported and used here instead.

**Review by hachiojidev — APPROVED** [review]

**Review by RabbitDoge — APPROVED** [review]

File: src/components/ModalInput/ModalInput.tsx [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+  const TranslateString = useI18n()
+  const isBalanceZero = max === '0' || !max
+
+  const displayBalance = isBalanceZero ? '0' : parseFloat(max).toFixed(4)
+
+  return (
+    <div style={{ position: 'relative' }}>
+      <StyledTokenInput isWarning={isBalanceZero}>
+        <Flex justifyContent="space-between" pl="16px">
+          <Text fontSize="14px">{withdrawal ? TranslateString(999, 'Unstake') : TranslateString(999, 'Stake')}</Text>
```
Comment (RichardPK) [inline src/components/ModalInput/ModalInput.tsx:77]: Thought of a better way of managing this to make it accept any string - don't merge just yet folks!

**Pull request merged**